urllib3/contrib/appengine.py

```diff
@@ -185,10 +185,14 @@ def _get_absolute_timeout(self, timeout):
         if timeout is Timeout.DEFAULT_TIMEOUT:
             return 5  # 5s is the default timeout for URLFetch.
         if isinstance(timeout, Timeout):
-            if timeout._read is not timeout._connect:
+            if timeout._read is not None or timeout._connect is not None:
                 warnings.warn(
                     "URLFetch does not support granular timeout settings, "
                     "reverting to total timeout.", AppEnginePlatformWarning)
+                if timeout.total is None:
+                    warnings.warn(
```

Comment:
I get that these two conditions are different, but is it worth drawing the distinction? I propose one of two options:

1. Keep it as just one message/branch and change it to "... reverting to total timeout (or default if not provided)."
2. If we want to keep two distinctive messages for the two cases, we should probably avoid printing 2 warnings in this scenario and catch that second branch as its own distinct scenario with a message that covers both conditions.